File: apps/DriftDate/shared/DriftCore/Sources/DriftCore/Models/Wave.swift
import Foundation

public struct Wave: Codable, Equatable, Identifiable, Sendable {
    public let id: UUID
    public let fromProfileId: UUID
    public let toProfileId: UUID
    public let layer: Layer
    public var status: WaveStatus
    public let createdAt: Date
    public var matchedAt: Date?

    public init(
        id: UUID = UUID(),
        fromProfileId: UUID,
        toProfileId: UUID,
        layer: Layer,
        status: WaveStatus = .pending,
        createdAt: Date = Date(),
        matchedAt: Date? = nil
    ) {
        self.id = id
        self.fromProfileId = fromProfileId
        self.toProfileId = toProfileId
        self.layer = layer
        self.status = status
        self.createdAt = createdAt
        self.matchedAt = matchedAt
    }
}

File: apps/DriftDate/android/core/src/main/java/com/americangroupllc/drift/core/models/Wave.kt
package com.americangroupllc.drift.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Wave(
    val id: String,
    @SerialName("from_profile_id") val fromProfileId: String,
    @SerialName("to_profile_id")   val toProfileId: String,
    val layer: Layer,
    val status: WaveStatus = WaveStatus.PENDING,
    @SerialName("created_at")  val createdAt: String,
    @SerialName("matched_at")  val matchedAt: String? = null,
)
